.github/workflows/linux-build-base.yml

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Linux Build
+
+on:
+  workflow_call:
+    inputs:
+      use-clang:
+        description: 'Use Clang to compile the project.'
+        default: ''
+        required: false
+        type: string
```

Comment:
Why not `boolean` ? Should be allowed https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype

Reply:
I will check but I wasn't sure if this wil "stringify" the value if use-clang is a boolean.
```
      USE_CLANG: "${{ inputs.use-clang }}"
```